examples/abi/res/abi.json

```diff
@@ -0,0 +1,119 @@
+{
+  "abi_schema_version": "0.1.0",
+  "metainfo": {
+    "name": "abi",
+    "version": "0.1.0",
+    "authors": [
+      "Near Inc <hello@nearprotocol.com>"
+    ]
+  },
+  "abi": {
+    "functions": [
+      {
+        "name": "add",
+        "is_view": true,
+        "is_init": false,
+        "params": [
+          {
+            "type_id": 0,
+            "serialization_type": "json"
+          },
+          {
+            "type_id": 0,
+            "serialization_type": "json"
+          }
+        ],
+        "result": {
+          "type_id": 0,
+          "serialization_type": "json"
+        }
+      },
+      {
+        "name": "add_callback",
+        "is_view": true,
+        "is_init": false,
+        "params": [],
+        "callbacks": [
+          {
+            "type_id": 1,
+            "serialization_type": "json"
+          },
+          {
+            "type_id": 1,
+            "serialization_type": "json"
+          }
+        ],
+        "callbacks_vec": {
+          "type_id": 2,
+          "serialization_type": "json"
+        },
+        "result": {
+          "type_id": 1,
+          "serialization_type": "json"
+        }
+      }
+    ],
+    "types": [
+      {
+        "id": 1,
+        "schema": {
+          "$ref": "#/definitions/DoublePair"
+        }
+      },
+      {
+        "id": 0,
+        "schema": {
+          "$ref": "#/definitions/Pair"
+        }
+      },
+      {
+        "id": 2,
+        "schema": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DoublePair"
+          }
+        }
+      }
+    ],
```

Comment:
What's the point of this section if it just references the schema? Can't we just ignore the references to this?

Reply:
> What's the point of this section if it just references the schema?

So what we are putting under `abi/root_schema` is called a root schema and it defines all custom types we are using in the contract. `schema` in this context refers to a subschema (which might be a better name for this json property), which **can** refer to something from the root schema. There are also cases where it can be just a primitive type (e.g. `{ "type": "integer" }`), a boolean value (e.g. `true` is a valid subschema meaning that all instances pass validation) among some other cases.

> Can't we just ignore the references to this?

I am not sure what you mean by this question, but if it helps I mentioned in Slack that type registry is actually no longer strictly necessary. We can just use subschemas directly inside `functions` since they are not that big. I am open to having a discussion on which way would be better.

Reply:
So it seems like the tradeoff is just more duplication if there are a lot of overlapping types, but more readable and less JSON when no overlap if inlining? Am I missing other considerations?

Reply:
Yeah, that's pretty much it

Reply:
As discussed in a meeting we decided to remove type registry for now and potential revisit this decision if the size of abi file becomes a big concern.